### ISSUE TITLE
Default values are completely removed for Avro Records with Union fields

### ISSUE THREAD

**Issue by enoren** [hortonworks/registry#298]
This issue appears to have been introduced with Issue #88 which resolves issues with default values and union fields, however when a record has a field of union type all default values are removed and set to null in the resultant schema. For instance, I have a schema such as:

```
{
      "namespace": "avro",
      "type": "record",
      "name": "UnionRecord",
      "fields": [
          {
              "name": "union_field",
              "type": ["null", "string"],
              "default": null
          }, {
              "name": "int_field",
              "type": "int",
              "default": 5
          }
      ]
}
```

When pulling this in from the schema registry the field 'int_field' no longer has the default value so any migrations from old schemas are failing as a result. The default value is in the schema when looking at the schema registry UI as well as while tracing through the code it is pulled down and shows up in the schema up until the schema is processed through the com.hortonworks.registries.schemaregistry.avro.AvroSchemaResolver#handleUnionFieldsWithNull method at which point the returned schema has lost the default value for the "int_field".

The actual issue appears to be related to this line:

```
updatedFields.add(new Schema.Field(field.name(),
                                                   fieldSchema,
                                                   field.doc(),
                                                   currentFieldTypeIsUnion ? NullNode.getInstance() : null,
                                                   field.order()));
```

However if I change it to

```
updatedFields.add(new Schema.Field(field.name(),
                                                   fieldSchema,
                                                   field.doc(),
                                                   currentFieldTypeIsUnion ? NullNode.getInstance() : field.defaultValue(),
                                                   field.order()));
```

everything appears to work correctly again.

**Comment by michaelandrepearce** [comment]
@enoren your logic change seems sensible for me, are you able to raise a PR?

**Comment by enoren** [comment]
@michaelandrepearce Done - https://github.com/hortonworks/registry/pull/301

**Comment by satishd** [comment]
Raised/merged #309 for adding tests
